### ISSUE TITLE
Highlighting is broken in some filetypes

### ISSUE THREAD

**Issue by binaryfunt** [atom/language-todo#82]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Highlighting has stopped working in JavaScript and Python files

### Steps to Reproduce

1. Open a JavaScript/Python file or an unsaved new file and set the language to JavaScript/Python
2. Type `TODO` in a comment

**Expected behavior:** `TODO` is highlighted, is in its own `<span>` with class `syntax--storage`

**Actual behavior:** `TODO` is grey/the same colour as the rest of the comment, is in the same `<span>` as the rest of the comment without class `syntax--storage`

### Versions

Atom: 1.29.0
Electron: 2.0.5
Chrome: 61.0.3163.100
Node: 8.9.3

### Additional information

Highlighting still works for C#

**Comment by 50Wliu** [comment]
Are you using the new Tree-Sitter Parsers option?

**Comment by binaryfunt** [comment]
Yes, I'm using Tree Sitter Parsers. If I turn it off then highlighting of `TODO` etc works again. Tree Sitter is still enabled in safe mode, so I didn't pick up on that.

**Comment by lgeiger** [comment]
Are there plans to support, tree-sitter grammars?

I'd be happy to help with this, since we're trying to achieve the same in the [hydrogen package](https://github.com/nteract/hydrogen). See https://github.com/atom/atom/issues/18196

**Comment by macdonaldtomw** [comment]
So, how can we disable tree-sitter until things get sorted out?  What version of atom should we be rolling back to?

**Comment by amangeot** [comment]
Thank you so much @macdonaldtomw, the new tree sitter option is really messing with syntax highlighting: TODO, JSDoc, JSX and more.

**Comment by rpgdev** [comment]
This was driving me insane. Thanks @macdonaldtomw!

**Comment by KvalheimRacing** [comment]
Thanks, this worked for me aswell!

**Comment by tehmaspc** [comment]
FYI: Still broken w/ Tree Sitter disabled in Terraform (.tf) files.

**Comment by agustinhaller** [comment]
Hi, any updates on the issue?

**Comment by KlfJoat** [comment]
I fully expect to be removed for spam, but this is a valid question that deserves to be here and to be answered...

Since this is an Atom-sponsored package, we expect it to work properly with Core option defaults.  This bug has just hit 1 year old.  Can we get an update?  Any visibility you can give us on the status of remediation?

Is there any workaround besides the one that disables other syntax highlighting to enable this?

**Comment by onebree** [comment]
I have tree-sitters disabled but still encountering this problem.

- Atom 1.41.0, macOS
- UI theme: One Dark
- Syntax theme: Base16 Tomorrow Dark
- Enabled / related packages:
  + atom/language-todo
  + atom/language-js
  + subtlegradient/language-javascript-jsx
- Affected grammer: JavaScript with JSX (`source.js.jsx`)

Under these conditions, inline comments (`//`) don't highlight todo-related keywords. But multi-line comments (`/* ... */`) do.

<img width="460" alt="image" src="https://user-images.githubusercontent.com/9038433/70166825-7ceafd00-1693-11ea-8a27-e9b4ebf5bb00.png">

**Comment by adamsoutar** [comment]
Is there any news on whether anyone is working on this or if it's even possibe to fix? This highlighting was a great atom feature, and tree-sitter parsers seem to be quite a lot faster, so a pain to loose.

**Comment by jonboiser** [comment]
I reported this on the language-javascript repo as well, since I think they need to modify that library to fix this.

Based on [these docs](https://flight-manual.atom.io/hacking-atom/sections/creating-a-grammar/#language-injection), language-javascript will need to add an injection point inside of all comment nodes, and then tell the parser to let the language-todo grammar take over, which would require setting an injectionRegex in this repository.

I'm not sure if this requires having a tree-sitter parser just for TODOs, or if it will still be compatible with the current textgrammar/regex grammar.

**Comment by hoytick** [comment]
Any word on an update for this or suggestions for how I can patch it on my machine? language-todo has the potential to be incredibly helpful, but I'd like it to work with Tree Sitter. I'm unfortunately not involved enough to be able to fix it myself.

**Comment by maximecb** [comment]
Still broken, missing this feature!

**Comment by jpm92** [comment]
No news in two years...I guess we can expect a fix soon, can we?

**Comment by KlfJoat** [comment]
Probably not.  GitHub is too busy purging repositories in response to illegal takedown requests (youtube-dl) to worry about this basic functionality.

**Comment by bookofportals** [comment]
I'm taking a look at fixing this. I wouldn't get your hopes up though. I'm a new contributor and I think this might actually be more complicated than it looks.

**Comment by maximecb** [comment]
Either way, thank you for taking a look at this @bookofportals, your effort is appreciated! :)

**Comment by johncmerfeld** [comment]
Whoever figures this out will be a real hero! I have to use the tree-sitter parser for TypeScript syntax to work, but it means I lose my TODOs, NOTEs, and FIXMEs.

**Comment by HunterKohler** [comment]
I'm not sure how atom highlights works on a deep level, nor tree-sitter parsers, but I am also having this issue.
Simply no highlighting.

**Comment by AlexWayfer** [comment]
> Interestingly, it works in the .less file itself, but not in python-files for me. Any ideas?

You can press Ctrl-Shift-I to open DevTools, then click arrow-like button at the top left corner, then select a part of code with `TODO` and see its selector. I guess, it has different classes, for example no `syntax--class` or `syntax--type` class. Not sure.

**Comment by AlexWayfer** [comment]
For me, in Ruby, there is no additional elements inside comments at all:

```
<span class="syntax--comment">## TODO: Check something</span>
```

So… probably it's language-specific-plugin issue, probably this work-around not for tree-sitter at all (I doubt).
